packages/commons/src/rate-limiter/redisRateLimiter.ts

```diff
@@ -0,0 +1,108 @@
+import { Redis } from "ioredis";
+import { TenantId } from "pagopa-interop-models";
+import {
+  IRateLimiterRedisOptions,
+  RateLimiterRedis,
+  RateLimiterRes,
+} from "rate-limiter-flexible";
+
+import { match, P } from "ts-pattern";
+import { Logger } from "../logging/index.js";
+import { RateLimiter, RateLimiterStatus } from "./rateLimiterModel.js";
+
+export function initRedisRateLimiter(config: {
+  limiterGroup: string;
+  maxRequests: number;
+  rateInterval: number;
+  redisHost: string;
+  redisPort: number;
+  timeout: number;
+}): RateLimiter {
+  const redisClient = new Redis({
+    enableOfflineQueue: false,
+    host: config.redisHost,
+    port: config.redisPort,
+    commandTimeout: config.timeout,
+    // ^ If a command does not return a reply within a set number of milliseconds,
+    // a "Command timed out" error will be thrown.
+  });
+  // TODO Should we set reconnectOnError to reconnect on specific errors?
+
+  const options: IRateLimiterRedisOptions = {
+    storeClient: redisClient,
+    keyPrefix: config.limiterGroup,
+    points: config.maxRequests,
+    duration: config.rateInterval / 1000, // seconds
+    // TODO should we recover on some erorrs?
+    // Documentation says its useful especially if using AWS ElastiCache with
+    // automatic failover disabled: https://github.com/redis/ioredis?tab=readme-ov-file#reconnect-on-error
+
+    // TODO figure out redis version we are using and set
+    // useRedisPackage accordingly: https://github.com/animir/node-rate-limiter-flexible/wiki/Redis#redis-package
+  };
+
+  // TODO do we need to allow traffict bursts and use the BURST_PERCENTAGE config?
+  // Seems to be supported but we need to set a burst points limit
+  // and also a burst rate interval.
+  // In Scala I see burstRequests = config.maxRequests * config.burstPercentage
+  // but what about the burst interval? Shoul we config.rateIntervalSeconds * config.burstPercentage?
```

Comment:
this seems a good enough approximation, we can always tune this up in case of need